src/guard.ts

```diff
@@ -37,8 +42,8 @@ export function createAuthGuard(app: App) {
   };
 }
 
-export async function authGuard(to: RouteLocation) {
+export async function authGuard(to: RouteLocation, redirectLoginOptions?: RedirectLoginOptions) {
```

Comment:
This breaks using the guard the way people should:

```
{
  path: '/profile',
  name: 'profile',
  component: Profile,
  beforeEnter: authGuard
}
```
We should ensure we do not break existing behavior.

Reply:
Not sure I follow on this one... the redirectLoginOptions is optional...

`redirectLoginOptions?:`

And the current api only shows authGuard having a "to" parameter. I'm assuming you're referring to when it's used in the above way and passed both "to" and "from" from Vue router? Still this api doesn't seem to do anything with the "from", and I'm not seeing how things break.

Are you saying that the function signature should be more like this:
authGuard(to: RouteLocation, from?: RouteLocation, redirectLoginOptions?: RedirectLoginOptions)

Sorry, but I'm unclear on this one...

Reply:
What I am saying is that our example we have (see playground/router.ts) no longer compiles:

<img width="979" alt="image" src="https://user-images.githubusercontent.com/2146903/218585662-2ba47a6e-2b6f-4f81-8c9a-89a03c562298.png">

It fails because Vue Router's Guard do not accept an optional RedirectLoginOptions. So we need to look into seeing if we can fix the types in such a way that Vue Router likes it, or remove the redirectLoginOptions alltogether

Reply:
> Are you saying that the function signature should be more like this:
> `authGuard(to: RouteLocation, from?: RouteLocation, redirectLoginOptions?: RedirectLoginOptions)`

No, that won't compile neither. 

I think we should have:

```ts
export async function authGuard(
  to: RouteLocation,
  from?: RouteLocation,
  next?: NavigationGuardNext,
  redirectLoginOptions?: RedirectLoginOptions
) {
  const auth0 = unref(auth0Client);

  return createGuardHandler(auth0, to, redirectLoginOptions);
}
```

Reply:
I know we do not use from and next, but a Guard in Vue is a function that takes 3 arguments, in a specific order: to, from and next. We can't just add `redirectLoginOptions` as a second argument.

Reply:
We will also need the exact same change to `createAuthGuard`.

Reply:
I got ya. I'll make the change in a bit and push an update. I didn't see the 3rd argument in the Vue docs per https://router.vuejs.org/guide/advanced/navigation-guards.html#optional-third-argument-next But I understand why this must be supported for backward compatibility.

Thank you for the clarification.

Reply:
No worries, thank you for the PR! 🚀